pkg/assets/apiserver/asset_apiserver.go

```diff
@@ -99,8 +104,10 @@ func NewAssetServerConfig(assetConfig oapi.AssetConfig) (*AssetServerConfig, err
 	if err := secureServingOptions.ApplyTo(genericConfig); err != nil {
 		return nil, err
 	}
-	genericConfig.SecureServingInfo.BindAddress = assetConfig.ServingInfo.BindAddress
-	genericConfig.SecureServingInfo.BindNetwork = assetConfig.ServingInfo.BindNetwork
+	genericConfig.SecureServingInfo.Listener, err = net.Listen(assetConfig.ServingInfo.BindNetwork, assetConfig.ServingInfo.BindAddress)
```

Comment:
we're opening a listener in the config constructor?

Reply:
if it's short term and this "config that is actually a running server" bit goes away, this is fine